externals/kyuubi-spark-sql-engine/pom.xml

```diff
@@ -190,6 +190,10 @@
                             <include>org.apache.curator:curator-client</include>
                             <include>org.apache.curator:curator-framework</include>
                             <include>org.apache.curator:curator-recipes</include>
+                            <include>org.apache.logging.log4j:log4j-1.2-api</include>
+                            <include>org.apache.logging.log4j:log4j-api</include>
+                            <include>org.apache.logging.log4j:log4j-core</include>
+                            <include>org.apache.logging.log4j:log4j-slf4j-impl</include>
```

Comment:
There are not required to be included, the engine which pick the right implementaion based on the runtime classes, right?

Reply:
these dependencies are needed for kyuubi-common

Reply:
`Logging` and `LogDivertAppender` will detect the runtime classes before invoking, it should be safe when running w/o log4j2 classes.

Reply:
are we good to remove this package?

Reply:
let me have a try

Reply:
```
  Cause: java.sql.SQLException: Failed to detect the root cause, please check /home/runner/work/incubator-kyuubi/incubator-kyuubi/kyuubi-server/target/work/runner/kyuubi-spark-sql-engine.log.0 at server side if necessary. The last 10 line(s) of log are:
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1039)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1048)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: java.lang.ClassNotFoundException: org.apache.logging.log4j.Level
	at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
	... 26 more
```

Reply:
Do you mean only remove `org.apache.logging.log4j:log4j-slf4j-impl`?

Reply:
Where is the `ClassNotFoundException`comes from, engine test or server test?

Reply:
I removed all the dependencies about log4j, then met the exception above, it happened in engine side.

Then I just removed `log4j-slf4j-impl` and it passed the GA at last.

Above exception should be expected.

Reply:
thanks @pan3793 , fixed